### PR TITLE
Added eslint:recommended to eslintrc.js

### PR DESCRIPTION
**Pull request by kendricktan**
**Motivation**

`eslint:recommended` enables checks for common problems. This helps us avoid common pitfalls.

**Summary**

Added `eslint:recommended` and globals (`web3`, `artifacts`).

**Issue(s)**

Fixes https://github.com/UMAprotocol/protocol/issues/1821

### CODE REVIEW

**Comment by adrianmcli** [comment]
side note, but this config will eventually need to be changed since we should be using the prettier config, not the plugin.

context: https://github.com/UMAprotocol/protocol/issues/1821#issuecomment-670387624

File: .eslintrc.js [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-  extends: ["plugin:prettier/recommended"],
+  extends: ["plugin:prettier/recommended", "eslint:recommended"],
   plugins: ["prettier", "mocha"],
   parserOptions: {
     ecmaFeatures: {
       jsx: true,
       modules: true
     }
   },
+  globals: {
```
Comment (nicholaspai) [inline .eslintrc.js:16]: What does this section do?

File: .eslintrc.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
 module.exports = {
   env: {
-    browser: true
+    browser: true,
```
Comment (nicholaspai) [inline .eslintrc.js:3]: OOC what does this do?

Comment (nicholaspai) [reply .eslintrc.js:3]: Nevermind answered my own questions here: https://eslint.org/docs/user-guide/configuring#specifying-globals

**Comment by kendricktan** [comment]
There seems to be [too many errors](https://app.circleci.com/pipelines/github/UMAprotocol/protocol/7795/workflows/315c9b7f-9832-4f37-af65-edef246ece41/jobs/56682/parallel-runs/0/steps/0-102) with the linting, i'll put this on hold for the time being

**Comment by daywiss** [comment]
can you explain the globals and how they work?

**Comment by kendricktan** [comment]
@daywiss 
Globals are so eslint doesn't warn you when you use them for custom runtimes (e.g. `artifacts`, `web3`)

**Comment by adrianmcli** [comment]
@daywiss you might find this instructive: https://eslint.org/docs/user-guide/configuring#specifying-globals

**Pull request closed without merging**